OsmAnd-java/src/main/java/net/osmand/GPXUtilities.java

```diff
@@ -1599,6 +1663,25 @@ public void updateWptPt(WptPt point, double lat, double lon, String description,
 			modifiedTime = System.currentTimeMillis();
 			pointsModifiedTime = modifiedTime;
 		}
+		
+		public void updateWptPt(WptPt point, double lat, double lon, String description, String name,
+		                        String category, String color, String iconName, String backgroundType, String address) {
```

Comment:
class is overloaded with lines of code, refactoring is needed and this method is not a good example that should be here especially wptName is not unique key!